src/components/Modals/ManageCollectionRoleModal/ManageCollectionRoleModal.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react'
+import { ModalNavigation, ModalActions, Button } from 'decentraland-ui'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import Modal from 'decentraland-dapps/dist/containers/Modal'
+
+import { isValid } from 'lib/address'
+import { getSaleAddress, setOnSale, isOnSale } from 'modules/collection/utils'
+import { Access, RoleType } from 'modules/collection/types'
+import { isEqual } from 'lib/address'
+import Role from './Role'
+import EmptyRole from './EmptyRole'
+import { Props, State } from './ManageCollectionRoleModal.types'
+import './ManageCollectionRoleModal.css'
+
+export default class ManageCollectionRoleModal extends React.PureComponent<Props, State> {
+  state: State = { roles: this.getStartRoles() }
+
+  getStartRoles() {
+    const { metadata } = this.props
+    const roles = new Set(metadata.roles)
+    roles.delete(this.getSaleAddress())
+    return Array.from(roles)
+  }
+
+  handleAddNewRole = () => {
+    const { roles } = this.state
+    this.setState({
+      roles: [...roles, undefined]
+    })
+  }
+
+  handleAddRole = (index: number, role: string) => {
+    if (this.isValidRole(role)) {
+      const roles = this.removeRoleAtIndex(index)
+      roles.push(role.toLowerCase())
+
+      this.setState({ roles })
+    }
+  }
+
+  handleRemoveRole = (role: string) => {
+    const { roles } = this.state
+    this.setState({
+      roles: roles.filter(_role => _role !== role)
+    })
+  }
+
+  handleCancelNew = (index: number) => {
+    this.setState({
+      roles: this.removeRoleAtIndex(index)
+    })
+  }
+
+  handleSubmit = () => {
+    const { collection, metadata } = this.props
+    const { roles } = this.state
+    const originalRoles = metadata.roles
+
+    const accessList: Access[] = []
+    for (const role of originalRoles) {
+      if (!roles.includes(role)) {
+        accessList.push({ address: role, hasAccess: false, collection })
+      }
+    }
+    for (const role of roles) {
+      if (!originalRoles.includes(role!)) {
+        accessList.push({ address: role!, hasAccess: true, collection })
+      }
+    }
+    this.setRoles(accessList)
+  }
+
+  removeRoleAtIndex(index: number) {
+    const { roles } = this.state
+    return [...roles.slice(0, index), ...roles.slice(index + 1)]
+  }
+
+  isValidRole(role: string) {
+    return isValid(role) && !this.state.roles.includes(role) && !isEqual(role, this.getSaleAddress())
+  }
+
+  getSaleAddress() {
+    const { wallet } = this.props
+    return getSaleAddress(wallet.networks.MATIC.chainId)
+  }
+
+  setRoles(accessList: Access[]) {
+    const { wallet, metadata, collection, onSetManagers, onSetMinters } = this.props
+    const { type } = metadata
+
+    // Restore the store access to the collection, removed on the first `state`
+    accessList.push(...setOnSale(collection, wallet, isOnSale(collection, wallet)))
```

Comment:
Following the comment above. We are "adding" again the store here if it should be turned on, and therefore the transaction has a _turn off and on_ of the store. `[..., store, store], [...,false, true]`

Reply:
Can we omit the store address if we are editing collaborators?